frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -54,25 +60,63 @@ class AddSubCategory extends React.Component<
       );
   };
 
-  onFinish = (values: any) => {
-    this.setState({ isLoading: true });
-    const subCategory = {
-      name: values.name,
-      category: {
-        id: values.category,
-      },
-    };
+  fetchSubCategoryDetails = (subCategoryId: string) => {
     axios
-      .post(
-        window.location.origin + '/core/academix/admin/sub-categories/',
-        subCategory
+      .get(
+        window.location.origin +
+          `/core/academix/sub-categories/${subCategoryId}`
       )
-      .then((res: AxiosResponse<SubCategory>) => {
-        if (res.status == 201) {
+      .then((result: AxiosResponse<SubCategory>) => {
+        if (result.status == 200) {
+          this.setState({ subcategory: result.data, isLoading: false });
+        } else {
+          throw new Error();
+        }
+      })
+      .catch((error) =>
+        handleApiError(
+          error,
+          'Something went wrong when trying to load the subcategory'
+        )
+      );
+  };
+
+  onFinish = (values: any) => {
+    let statusCode: number, method: Method, url, subCategory;
+    if (this.subCategoryId == null) {
```

Comment:
Instead of checking subcategory is null, define the component type ("add" or "edi") as soon as the component get mounted

Reply:
fixed